src/dialect/mod.rs

```diff
@@ -141,4 +192,32 @@ mod tests {
         assert!(dialect_of!(ansi_holder is GenericDialect | AnsiDialect));
         assert!(!dialect_of!(ansi_holder is GenericDialect | MsSqlDialect));
     }
+
+    #[test]
+    fn test_dialect_from_str() {
+        assert!(parse_dialect("generic").is::<GenericDialect>());
+        assert!(parse_dialect("mysql").is::<MySqlDialect>());
+        assert!(parse_dialect("MySql").is::<MySqlDialect>());
+        assert!(parse_dialect("postgresql").is::<PostgreSqlDialect>());
+        assert!(parse_dialect("postgres").is::<PostgreSqlDialect>());
+        assert!(parse_dialect("hive").is::<HiveDialect>());
+        assert!(parse_dialect("sqlite").is::<SQLiteDialect>());
+        assert!(parse_dialect("snowflake").is::<SnowflakeDialect>());
+        assert!(parse_dialect("SnowFlake").is::<SnowflakeDialect>());
+        assert!(parse_dialect("MsSql").is::<MsSqlDialect>());
+        assert!(parse_dialect("clickhouse").is::<ClickHouseDialect>());
+        assert!(parse_dialect("ClickHouse").is::<ClickHouseDialect>());
+        assert!(parse_dialect("bigquery").is::<BigQueryDialect>());
+        assert!(parse_dialect("BigQuery").is::<BigQueryDialect>());
+        assert!(parse_dialect("ansi").is::<AnsiDialect>());
+        assert!(parse_dialect("ANSI").is::<AnsiDialect>());
```

Comment:
I wish there were a way for us to throw a compile error if one of the dialects isn't tested here (or alternatively, loop through a list of dialects and roundtrip them or something to make sure each one is represented).

Reply:
I agree that would be great. 

The only thing I can think of is to make an enum or something with all the dialects 🤔